### ISSUE TITLE
GDExtension API dump wrong type for enum VoxelInstancer::UpMode

### ISSUE THREAD

**Issue by Superbelko** [Zylann/godot_voxel#648]
**Describe the bug**
GDExtension API dumps wrong enum, `VoxelInstancer::UpMode` (`voxel/terrain/instancing/voxel_instancer.h`)  is declared in `voxel/terrain/instancing/up_mode.h` but seems it is not exported correctly, this "breaks" the GDExtension binding generators and I have to manually fix generated code.

**To Reproduce**
1. build Godot with Voxel engine module
2. dump GDExtension API `godot --dump-extension-api`
3. See generated JSON 

**Expected behavior**
From generated bindings It seems the best fix would be to properly export that enum in `VoxelInstancer` class, or make it global enum.
For example `VoxelMesherCubes::ColorMode` has similar layout and works fine.

---

 The only difference between these two seems to be that `VoxelMesherCubes` has ColorMode enum defined in class, but VoxelInstancer is using external declaration.

---

**Screenshots**
dumped API looks like this, it has enum in there but in methods it has wrong type, just to be clear there is no 'voxel' anywhere in generated bindings, that's the problem, it should be VoxelInstancer instead like for example in `VoxelMesherCubes::ColorMode`.

```json
		{
			"name": "VoxelInstancer",
			"is_refcounted": false,
			"is_instantiable": true,
			"inherits": "Node3D",
			"api_type": "core",
			"constants": [
				{
					"name": "MAX_LOD",
					"value": 8
				}
			],
			"enums": [
				{
					"name": "UpMode",
					"is_bitfield": false,
					"values": [
						{
							"name": "UP_MODE_POSITIVE_Y",
							"value": 0
						},
						{
							"name": "UP_MODE_SPHERE",
							"value": 1
						}
					]
				},
```

```json
				{
					"name": "set_up_mode",
					"is_const": false,
					"is_vararg": false,
					"is_static": false,
					"is_virtual": false,
					"hash": 303930206,
					"arguments": [
						{
							"name": "mode",
							"type": "enum::voxel.UpMode"
						}
					]
				},
```

**Environment**
- OS: Windows 11 x64
- Godot version 4.3 beta1
- Module version fd340cf8097a50f03dea1be0e4c93511fc0e02c9

**Comment by Zylann** [comment]
> The only difference between these two seems to be that VoxelMesherCubes has ColorMode enum defined in class, but VoxelInstancer is using external declaration.

Damn, that's so annoying. It works fine in Godot, why not GodotCpp? It even shows that the enum is in the class, so surely it knows, and why is there that "voxel" in there? When I call `BIND_ENUM_CONSTANT(UP_MODE_POSITIVE_Y)` there is no namespace specified anywhere. Where does it come from??

I moved that enum because it forced me to `#include` a class that could otherwise be forward-declared. And having to alias every enum member creates code repetition...

**Comment by Zylann** [comment]
Attempted fix in 6ebf995ee6ccaf07b11121447485d219b1a558f1

**Comment by Superbelko** [comment]
Works for me now.

Ignore that global enums thing, it seems this stuff is isolated and is for core types only (in `godot/core/core_constants.cpp`)
Yeah, unfortunately there isn't much can be done, but at least it works.
